### PR TITLE
Use cita database chain

### PR DESCRIPTION
**Pull request by kaikai1024**
### Description of the Change

Rebase the branch: remove-snapshot. please review https://github.com/cryptape/cita/pull/700 first

### CODE REVIEW

File: cita-chain/src/main.rs [hunk 1]
```diff
@@ -73,13 +73,14 @@ extern crate libproto;
 extern crate cita_logger as logger;
 #[macro_use]
 extern crate util;
-extern crate db as cita_db;
+
+extern crate cita_database as cita_db;
```
Comment (leeyr338) [inline cita-chain/src/main.rs]: You don't need to use `extern` in Edition 2018?

File: cita-chain/src/main.rs [hunk 2]
```diff
@@ -133,8 +134,8 @@ fn main() {
 
     let nosql_path = DataPath::nosql_path();
     trace!("nosql_path is {:?}", nosql_path);
-    let db_config = DatabaseConfig::with_columns(db::NUM_COLUMNS);
-    let db = Database::open(&db_config, &nosql_path).unwrap();
+    let db_config = DatabaseConfig::with_category_num(db::NUM_COLUMNS);
```
Comment (leeyr338) [inline cita-chain/src/main.rs]: `COLUMN`  is a proper noun from **Storage**, so I think the name `with_columns` is better than `with_category_num`.

File: cita-chain/src/main.rs [hunk 3]
```diff
@@ -133,8 +134,8 @@ fn main() {
...
+    let db = RocksDB::open(&nosql_path, &db_config).unwrap();
```
Comment (leeyr338) [inline cita-chain/src/main.rs]: `expect` is better than `unwrap`, we need to know what happen when the system crash.

File: cita-chain/src/main.rs (hunk 1 above)
Comment (kaikai1024) [reply cita-chain/src/main.rs]: How the use `as` without `extern`?

File: cita-chain/src/main.rs (hunk 2 above)
Comment (kaikai1024) [reply cita-chain/src/main.rs]: You can check the cita-database. All the interface use the category as the params.
category is a abstract of column. this column is the noun fo rocksdb.

We can discuss it in a issue of cita-database.

File: cita-chain/src/main.rs (hunk 3 above)
Comment (kaikai1024) [reply cita-chain/src/main.rs]: It's anothere problem.

I just replace the old logic with cita-databse

File: cita-chain/src/main.rs (hunk 2 above)
Comment (leeyr338) [reply cita-chain/src/main.rs]: ok.

File: cita-chain/src/main.rs (hunk 3 above)
Comment (leeyr338) [reply cita-chain/src/main.rs]: ok

File: cita-chain/src/main.rs (hunk 1 above)
Comment (leeyr338) [reply cita-chain/src/main.rs]: Have you try:
```rust
use db as cita_db
```
?

File: cita-chain/src/main.rs (hunk 3 above)
Comment (kaikai1024) [reply cita-chain/src/main.rs]: #704

File: cita-chain/src/main.rs (hunk 1 above)
Comment (kaikai1024) [reply cita-chain/src/main.rs]: Wait a minute.

File: cita-chain/Cargo.toml [hunk 4]
```diff
@@ -21,7 +21,7 @@ jsonrpc-types = { git = "https://github.com/cryptape/cita-common.git", branch =
 core = { path = "./core" }
 common-types = { path = "./types" }
 proof = { git = "https://github.com/cryptape/cita-common.git", branch = "develop" }
-db = { git = "https://github.com/cryptape/cita-common.git", branch = "develop" }
+cita_db = { git = "https://github.com/cryptape/cita-database.git", branch = "parity-rocksdb", package = "cita-database" }
```
Comment (kaikai1024) [inline cita-chain/Cargo.toml]: @leeyr338  here

**Pull request merged**